### PR TITLE
Revision of material/medium index access; Simplifications in Base/Det…

### PR DESCRIPTION
**Pull request by sawenzel**
…ector

 * remove virtual keyword from some functions
 * do not use pointers to containers
 * use map instead of vector for index lookup, since the index might be sparse
 * rename getMaterial ---> getMaterialID for better clarity
 * remove getter to underlying container (not used)
 * corresponding changes in EMCAL, TOF, TPC detector classes

### CODE REVIEW

**Comment by sawenzel** [comment]
@mfasDa , @wiechula : Can you please review? This is just a stability/cleanup commit. No new functionality is added.

**Comment by mfasDa** [comment]
I have looked at the code and from EMCAL side it is fine.

File: Detectors/Base/include/DetectorsBase/Detector.h [hunk 1]
```diff
@@ -95,18 +109,18 @@ class Detector : public FairDetector
 
 
     /// Mapping of the ALICE internal material number to the one
-    /// automatically assigned by geant.
-    /// This is required for easily being able to copy the geometry setup
+    /// automatically assigned by geant/TGeo.
+    /// This is required to easily being able to copy the geometry setup
     /// used in AliRoot
-    std::unique_ptr<std::vector<int>> mMapMaterial; //!< material mapping
+    std::map<int, int> mMapMaterial; //! material mapping
```
Comment (wiechula) [inline Detectors/Base/include/DetectorsBase/Detector.h]: For doxygen I think the comment should be //!< to make clear the comment belongs to the line before not after. Same below.

Comment (sawenzel) [inline Detectors/Base/include/DetectorsBase/Detector.h]: ok. Didn't know about this.

Comment (wiechula) [inline Detectors/Base/include/DetectorsBase/Detector.h]: See here:
http://www.stack.nl/~dimitri/doxygen/manual/docblocks.html#memberdoc

**Pull request merged**